### ISSUE TITLE
switch from iText to OpenPDF?

### ISSUE THREAD

**Issue by goekay** [opensagres/xdocreport#258]
this has been mentioned in the issue #174, but there the focus of discussion was something else. how about switching to https://github.com/LibrePDF/OpenPDF?

problem with itext 2.1.7 is that it's old (from 2009). openpdf is a fork of itext 4 with a friendlier license than the current itext. it uses the same namespace and APIs as itext, so the switch only requires some maven dependency changes.

**Comment by angelozerr** [comment]
@pascalleclercq what do you think about that?

**Comment by amedee** [comment]
What is referred to as iText 4, was never released to Maven Central by iText Software. It was just an internal svn tag used for syncing up the Java and C# versions. See https://github.com/LibrePDF/OpenPDF/issues/18 for details.
tl;dr: "iText 4" = iText 2.1.7 + a few commits that were made in July-November 2009.

None of the commits from the period 2010-2018 are in "iText 4", and consequently also not in OpenPDF. Because of the license change in iText, none of the changes in iText can make it to OpenPDF, unless OpenPDF changes their license to AGPLv3. If OpenPDF would take a bugfix from iText 5 and cherry-pick that to OpenPDF, it would be a copyright infringement. They would have to rewrite that bugfix in a blackbox approach.
There is a very high risk that OpenPDF indeed has such copyright infringements.

**Comment by keesvandieren** [comment]
#387  is a duplicate of this issue.

**Comment by andreasrosdal** [comment]
I would support the proposal to switch to OpenPDF, as it is maintained, and has a dual LGPL/MPL license.

OpenPDF is an open source project where we welcome contributions from the community. It has a dual LGPL/MPL license, and is used in accordance with this license in the original source code published by iText on Sourceforge. We have done a code license review and it is clear that OpenPDF uses the source code in accordance with the LGPL/MPL license. (amede, if there is copied code in the most recent release or source code, please tell me, and I will remove it. However, no such code has been found or reported.)

**Comment by angelozerr** [comment]
2.0.3 has been released and provides an openpdf support.
